File: 通讯协议/ProtoGen/output/handlers/csharp_all/StealHandler.cs
/*-----------------------------------------------
 *本文件由代码生成器自动生成，
 *千万不要修改本文件的任何代码，
 *修改的的任何代码都会被覆盖掉！
 --------------------------------------------------*/

using System.IO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Pomelo.DotNetClient;

namespace pomelo.area
{
	public class StealHandler
	{
		private PomeloClient _socket;

		public StealResponse lastStealResponse { get { return _socket.GetLastResponse<StealResponse>(); } }
		static StealHandler()
		{
			EventTypes.RegistRequestType("area.stealHandler.stealRequest", typeof(StealRequest), typeof(StealResponse));
		}
		public StealHandler(PomeloClient socket)
		{
			this._socket = socket;

		}
		public void stealRequest(int c2s_id,Action<PomeloException,StealResponse> cb,object option = null)
		{
			var request = new StealRequest();
			request.c2s_id= c2s_id;
			_socket.request<StealResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as StealResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}

	}


}

File: 通讯协议/ProtoGen/output/handlers/lua/StealHandler.lua
-------------------------------------------------
-- 本文件由代码生成器自动生成，
-- 千万不要修改本文件的任何代码，
-- 修改的的任何代码都会被覆盖掉！
----------------------------------------------------

local Socket = require "Xmds.Pomelo.LuaGameSocket"
require "base64"
require "stealHandler_pb"

--------------------------------Pomelo-------------------------------
Pomelo = Pomelo or {}

--------------------------------StealHandler-------------------------------
Pomelo.StealHandler = {}

local function stealRequestEncoder(msg)
	local input = stealHandler_pb.StealRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function stealRequestDecoder(stream)
	local res = stealHandler_pb.StealResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.StealHandler.stealRequest(c2s_id,cb,option)
	local msg = {}
	msg.c2s_id = c2s_id
	Socket.OnRequestStart("area.stealHandler.stealRequest", option)
	Socket.Request("area.stealHandler.stealRequest", msg, function(res)
		if(res.s2c_code == 200) then
			Pomelo.StealHandler.lastStealResponse = res
			Socket.OnRequestEnd("area.stealHandler.stealRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] area.stealHandler.stealRequest decode error!!"
			end
			Socket.OnRequestEnd("area.stealHandler.stealRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, stealRequestEncoder, stealRequestDecoder)
end




--------------------------------Pomelo-------------------------------
return Pomelo
